trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/expressiontranslation/IntegerTranslation.java

```
@@ -802,4 +783,88 @@ protected Pair<Expression, Expression> constructOverflowCheckForLeftShift(final
 			final ExpressionResult exprResult) {
 		return constructOverflowCheck(loc, resultType, exprResult.getLrValue().getValue());
 	}
+
+	@Override
+	public Expression roundToIntegral(final ILocation loc, final Expression argument, final CPrimitive type,
+			final Expression roundingMode) {
+		throw new UnsupportedOperationException(NOT_IMPLEMENTED);

```

Would it (in the future, not this PR) make sense to implement some of these functions with overapproximations as well? Or do we prefer to fail early if these are used, e.g. because we assume only bitvector mode will have a chance to verify such programs? (CC @danieldietsch ?)

---

Oh I think in particular for float we could have various different implementations, also with overapproximations. I don't think we want to fail early (or: at least not unconditionally), but rather always have the option of, e.g., havoc + assume inside type range + overapprox annotation.

---

In principle, it could be helpful, e.g. if a call to such a function does not really influence the correctness of the program or if a violation already occurs before a call to such a function (which seems both not too likely), since we already crash here before running the actual verification.
This PR makes this easier, only a relatively small refactoring should be required to return an `ExpressionResult` that contains the overapproximation instead of an `Expression` for those methods.

---

>  I don't think we want to fail early (or: not unconditionally, it should be an option)

Maybe another argument for some splitting: `IntegerExpressionTranslation` could select a different `IFloatingPointHandler` implementation based on a setting :)